platform/mbed_retarget.h

```diff
@@ -66,29 +73,63 @@ namespace mbed {
 
 class FileHandle;
 class DirHandle;
-std::FILE *mbed_fdopen(FileHandle *fh, const char *mode);
+
+/** Bind an mbed FileHandle to a POSIX file descriptor */
+int bind_to_fd(mbed::FileHandle *fh);
+
+/** Unbind an mbed FileHandle from a POSIX file descriptor
+ * Intended for internal use - normally unbinding happens
+ * as part of close() or fclose().
+ */
+int unbind_from_fd(int fd, mbed::FileHandle *fh);
```

Comment:
If these are intended for internal use, do we want them to be part of doxygen? Removing the second asterisk should prevent them from appearing on the docs site